### PR TITLE
pkg/termios: fix darwin regression from previous commit, add tests

### PR DESCRIPTION
**Pull request by bradfitz**
I broke darwin support in the just-merged #2593.

To atone for my sins, this PR adds some more test coverage (in addition to fixing darwin).

### CODE REVIEW

**Comment by paulmenzel** [comment]
Small typo in description: ad*d*ition

**Comment by bradfitz** [comment]
> Small typo in description: addition

Whoops, thanks. Fixed.

File: pkg/termios/termios_test.go [hunk 1]
```diff
@@ -267,3 +269,53 @@ func TestSet(t *testing.T) {
...
+		"openbsd/amd64",
+	}
+	// As of Go 1.19+, running "go" selects the right one:
+	//
+	// https://go.dev/doc/go1.19
+	//
+	// "go test and go generate now place GOROOT/bin at the
+	// beginning of the PATH used for the subprocess, so tests and
+	// generators that execute the go command will resolve it to
+	// same GOROOT"
```
Comment (hugelgupf) [inline pkg/termios/termios_test.go:304]: Wow, that's cool. I had to fix that myself in the past... really annoying. [see e.g.](https://github.com/u-root/gobusybox/blob/46b2883a7f908fe80f7a9580e136b8d256856c47/src/pkg/golang/build.go#L127)

**Review by hugelgupf — DISMISSED** [review]

File: pkg/termios/termios_test.go [hunk 2]
```diff
@@ -267,3 +269,53 @@ func TestSet(t *testing.T) {
...
+// Compare:
+//
+//	GOOS=linux   GOARCH=amd64 go doc golang.org/x/sys/unix.Termios
+//	GOOS=darwin  GOARCH=amd64 go doc golang.org/x/sys/unix.Termios
+//	GOOS=openbsd GOARCH=amd64 go doc golang.org/x/sys/unix.Termios
+//
+// It's all a mess.
+//
+// This at least makes sure that the package compiles on all platforms.
+func TestCrossCompile(t *testing.T) {
```
Comment (hugelgupf) [inline pkg/termios/termios_test.go:284]: Can you also add 

testutil.SkipIfInVMTest(t)

inside our VM infra, the go tool is not available.

Comment (bradfitz) [reply pkg/termios/termios_test.go:284]: Done. I thought that CIFuzz was failing even on my prior change for unrelated reasons, though, even from before I added a test. So I'd just ignored its failure this time.

**Review by hugelgupf — APPROVED** [review]

**Pull request merged**